src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -3200,20 +3200,23 @@ void mvnw(Register Rd, Register Rm,
   }
 
   // SVE dup imm
-  void sve_dup(FloatRegister Zd, SIMD_RegVariant T, int imm8) {
+  void sve_dup(FloatRegister Zd, SIMD_RegVariant T, int imm16) {
     starti;
     assert(T != Q, "invalid size");
     int sh = 0;
-    if (imm8 <= 127 && imm8 >= -128) {
+    unsigned imm = imm16;
+    if (imm16 <= 127 && imm16 >= -128) {
       sh = 0;
-    } else if (T != B && imm8 <= 32512 && imm8 >= -32768 && (imm8 & 0xff) == 0) {
+    } else if (T != B && imm16 <= 32512 && imm16 >= -32768 && (imm16 & 0xff) == 0) {
       sh = 1;
-      imm8 = (imm8 >> 8);
+      imm = (imm >> 8);
```

Comment:
Hi @theRealAph , thanks for looking at this PR, and thanks for your comment here. Yes, I agree that the compilers we know like GCC/LLVM can make sure the behavior is defined on AArch64. And actually we didn't met any issues here. However, I'm not quite sure whether other compilers can guarantee it. This is just used to avoid the undefined behavior in future.  So do you think we need to fix it here? I can abandon this patch if this is not valuable. Thanks!

Reply:
The compilers do guarantee it. Here's GCC, for example:

4.5 Integers

GCC supports only two’s complement integer types, and all bit patterns are ordinary values. 

Bitwise operators act on the representation of the value including both the sign and value bits, where the sign bit is considered immediately above the highest-value value bit. Signed ‘>>’ acts on negative numbers by sign extension.

Reply:
Thanks a lot for the explanation. I will close this PR as it is no need to modify it.